### ISSUE TITLE
Remove 1 sat hard limit

### ISSUE THREAD

**Issue by zpxp** [bitcoin-sv/bitcoin-sv#261]
We have a critical business requirement that requires data tracked via utxos but cannot reclaim the locked sat later. This can only be achieved with zero sat utxos. Putting in a single sat will drain wallets and allow hackers to skim sats off multisig utxos. Please remove.

**Comment by baryon** [comment]
Change your business!

**Comment by zpxp** [comment]
What happened to "the protocol must be set in stone"? Or is it okay when craig decides to change the protocol?

**Comment by baryon** [comment]
Dr. CSW has said that bitcoin was set in stone from the day it was born, and even he can't change that. 

0STAS is an implementation bug.
Don't make similar proposals again

**Comment by zpxp** [comment]
It used to work, now it doesn't, that's a protocol change

**Comment by Danconnolly** [comment]
I'm sorry but we're not going to do that.

**Comment by zpxp** [comment]
So the all wise and powerful devs at the bitcoin association can change the protocol at their discretion, while putting up a public veneer of "set in stone" and "build on BSV, we wont break your existing software". Well guess what, existing software has been broken and it appears the bsv ivory tower is content with that. This will be a PR disaster for bsv.

**Comment by baryon** [comment]
no one can change stone. Bitcoin is a cash system, not a data system.

**Comment by zpxp** [comment]
@baryon As someone who is trying to run a bsv business, this should greatly concern you, that the association can make breaking changes to protocol at any time.

**Comment by deanmlittle** [comment]
> Dr. CSW has said that bitcoin was set in stone from the day it was born, and even he can't change that.
> 
> 0STAS is an implementation bug. Don't make similar proposals again

It was not an implementation bug. Craig forgot to do his homework on Satoshi's SourceForge commits and accidentally admitted to the world that he is, without a doubt, not the one who wrote the code you speak of, despite it being Satoshi code. Anyone with even a rudimentary understanding of C++ who reads the code comments Satoshi made, along with the perfectly executed logic directly below them, could only come to one conclusion: Satoshi fully intended for 0 sat utxos to be both created and spent. It was not a mistake, as the code comments point out, and disabling it is a deviation from the original protocol. I'm sorry you have been conned by this man's lies, but this is an undeniable fact.

![image](https://user-images.githubusercontent.com/10921578/167377985-26304c8b-fc48-4bd2-b799-86c7b1259b79.png)

**Comment by baryon** [comment]
I don't care if Dr. CSW is Satoshi Nakamoto. I believe what the Bitcoin whitepaper does not include 0. Because it is a cash system, not a data system. The biggest misdirection in the last two years is to misinterpret big blocks as big data.

**Comment by zpxp** [comment]
> I don't care if Dr. CSW is Satoshi Nakamoto. I believe what the Bitcoin whitepaper does not include 0. Because it is a cash system, not a data system. The biggest misdirection in the last two years is to misinterpret big blocks as big data.

By that logic we should delete all OP_RETURN and OP_PUSHDATA codes then, no?

**Comment by baryon** [comment]
> > I don't care if Dr. CSW is Satoshi Nakamoto. I believe what the Bitcoin whitepaper does not include 0. Because it is a cash system, not a data system. The biggest misdirection in the last two years is to misinterpret big blocks as big data.
> 
> By that logic we should delete all OP_RETURN and OP_PUSHDATA codes then, no?

A receipt for a cash transaction or digital product can also be stored on the blockchain. Data are only meaningful if they relate to cash transactions.

**Comment by baryon** [comment]
The right thing to raise dust limit, not zero. And should strongly oppose MAPI, it is a big data backdoor

**Comment by zpxp** [comment]
> The right thing to raise dust limit, not zero. And should strongly oppose MAPI, it is a big data backdoor

Doing that would change the protocol.

> no one can change stone. Bitcoin is a cash system, not a data system.

So much for not changing stone

**Comment by baryon** [comment]
I don't see anything that needs to change. 

But apparently the miners didn't realize that too low a Dust limit could lead to systemic risk. Node developers have removed the restrictions.

**Comment by SoiledCold5** [comment]
I support zpxp

**Comment by torusJKL** [comment]
@zpxp I recommend you contact the GorillaPool.
@MineLikeAnApe on Twitter.

They might be able to help you by including your 0 sat outputs.

From what I understand once they are in a block they are still regarded as valid.

**Comment by zpxp** [comment]
@torusJKL Thanks, I've had preliminary discussions with the guys over there. Lets see if they can sort it out.

**Comment by jadwahab** [comment]
> We have a critical business requirement that requires data tracked via utxos but cannot reclaim the locked sat later. This can only be achieved with zero sat utxos. Putting in a single sat will drain wallets and allow hackers to skim sats off multisig utxos. Please remove.

@zpxp why can't you reclaim the locked sat later? Sounds like a fundamental issue with the way your system is set up...

**Comment by mpapec** [comment]
> > I don't care if Dr. CSW is Satoshi Nakamoto. I believe what the Bitcoin whitepaper does not include 0. Because it is a cash system, not a data system. The biggest misdirection in the last two years is to misinterpret big blocks as big data.
> 
> By that logic we should delete all OP_RETURN and OP_PUSHDATA codes then, no?

I'm failing to see connection between spendable 0 sat utxo, and suggestion above.

**Comment by zpxp** [comment]
> > We have a critical business requirement that requires data tracked via utxos but cannot reclaim the locked sat later. This can only be achieved with zero sat utxos. Putting in a single sat will drain wallets and allow hackers to skim sats off multisig utxos. Please remove.
> 
> @zpxp why can't you reclaim the locked sat later? Sounds like a fundamental issue with the way your system is set up...

Well of course its possible, but would require tracking those utxos in a database after the client notifies us of it's existence. It's far more inefficient.

**Comment by zpxp** [comment]
But the point it shouldn't have broke in the first place. BSV's only real selling point was "build on us and existing code won't be broken". I can adapt and change the code, sure, but now a precedent is set where at any time, the "stable" protocol can be changed by a select group's discretion. Kind of the opposite of BSV's only real selling point.

**Comment by KaKeimei** [comment]
Interesting

**Comment by Danconnolly** [comment]
The protocol is fixed in stone and will not change.

The software is continually evolving.

**Comment by zpxp** [comment]
> The Bitcoin rules are fixed in stone and will not change.
> 
> The software is continually evolving.

Nice misdirection by saying we didn't change the protocol, we only changed the code. What implements the bitcoin rules? The software.  Yet for some reason no-one can seem to address the fact that the protocol did change, they only ignore it, stonewall it, or make completely illogical and pointless arguments.

**Comment by Danconnolly** [comment]
You might be referring to two things there:

1. The original response I wrote said "The protocol is fixed...". I dont like the word "protocol". It is used in the Bitcoin space in confusing ways to refer to multiple things. For me, it refers to the network protocol which the node software uses to communicate with each other. This protocol has changed in the past (Compact Blocks) and will change in the future (SBT). This protocol is not "Bitcoin". That's why I changed it to "The Bitcoin rules are fixed ...".
2. You might think I'm being obtuse by saying that the Bitcoin rules have not changed. But that is the whole point. Anyone can write any software and as long as it follows the Bitcoin rules, it will be Bitcoin and the transactions it produces *can* be accepted in the blockchain. I said "can" and not "will". You may need to find a miner to include them in blocks. 
 
The bottom line is that if you want to use Bitcoin in a way that follows the rules but is not accepted by this node software, then please do go ahead and write your own node software. You can even fork this software, as long as you follow the terms of the license.

**Comment by torusJKL** [comment]
@zpxp 
The distinction Danconnolly is making is very important to understand.
There are the common Bitcoin rules (set in stone with known exceptions) and there are the rules a node software implements.

As long as the rules implemented by the node software are allowed by the Bitcoin rules it is not a change.
It is simply a design decision made by that specific node implementation.

It would have been a change in the rules if the node would reject blocks with 0 sat outputs but AFAIK that is not the case.
It simply doesn't accept them when broadcasted to it outside of a block.

Now, it might be frustrating if something had worked by default with most miners and doesn't anymore after they upgraded to the latest version.
But it is perfectly fine to contact miners and ask them to include your transactions despite this new default behavior.
They will need to do some adjustments but I'm sure they are capable and willing if they see an economic incentive in it.

**Comment by ContrarianUnderscoreUnderscore** [comment]
@Danconnolly , does this mean BSV will not allow signed transactions from 2009 as promised? What about OP_VER ? That was "scheduled to be re-enabled in the Chronicle release". Would you consider that addition a change in the "Bitcoin rules"? 

What about the change to reassign coins due to a "court order" that nChain's Chief Scientist claimed in a lawsuit was coming to BSV? Will that change any existing "Bitcoin rules"? 

In general, could you be more precise about what you mean by "the Bitcoin rules" and what it means to be "fixed"? Does that mean that a hard forking change (for instance, allowing transactions signed with the original sighash algorithm) would be acceptable or not? What about soft forking changes, like the one where you changed the original Bitcoin rules to disallow non-push opcodes in the scriptSig?

**Comment by mpapec** [comment]
> @Danconnolly , does this mean BSV will not allow signed transactions from 2009 as promised? What about OP_VER ? That was "scheduled to be re-enabled in the Chronicle release". Would you consider that addition a change in the "Bitcoin rules"?
> 
> What about the change to reassign coins due to a "court order" that nChain's Chief Scientist claimed in a lawsuit was coming to BSV? Will that change any existing "Bitcoin rules"?
> 
> In general, could you be more precise about what you mean by "the Bitcoin rules" and what it means to be "fixed"? Does that mean that a hard forking change (for instance, allowing transactions signed with the original sighash algorithm) would be acceptable or not? What about soft forking changes, like the one where you changed the original Bitcoin rules to disallow non-push opcodes in the scriptSig?

Take it easy, you'll know everything when a time is right. :-)

**Comment by ContrarianUnderscoreUnderscore** [comment]
> Take it easy, you'll know everything when a time is right. :-)

@mpapec, I don't know what you mean by this. It doesn't answer any of my questions and is, frankly, unprofessional and nonsensical. 

"The Bitcoin rules are fixed in stone and will not change" is a straightforward-seeming comment, but it has nothing close to a formal definition to my knowledge. I'm simply asking for clarification.

**Comment by mpapec** [comment]
> > Take it easy, you'll know everything when a time is right. :-)
> 
> @mpapec, I don't know what you mean by this. It doesn't answer any of my questions and is, frankly, unprofessional and nonsensical.
> 
> "The Bitcoin rules are fixed in stone and will not change" is a straightforward-seeming comment, but it has nothing close to a formal definition to my knowledge, especially since further changes were promised at some point. I'm simply asking for clarification.

Could you explain what business exactly are you running on bsv and why do you care? 
Thank you in advance. :-)

**Comment by ContrarianUnderscoreUnderscore** [comment]
@mpapec , please stay out of the discussion if you're not going to contribute.

**Comment by mpapec** [comment]
> @mpapec , please stay out of the discussion if you're not going to contribute.

@ContrarianUnderscoreUnderscore please do not troll concern if it's none of your business. :-)
